bitcoin/src/network/constants.rs

```diff
@@ -408,4 +495,20 @@ mod tests {
             assert_eq!(back, network);
         }
     }
+
+    #[test]
+    fn magic_from_str() {
+        let known_network_magic_strs = [
+            ("0xf9beb4d9", Network::Bitcoin),
+            ("0x0b110907", Network::Testnet),
+            ("0xfabfb5da", Network::Regtest),
+            ("0x0a03cf40", Network::Signet),
+        ];
+
+        for (magic_str, network) in &known_network_magic_strs {
+            let magic: Magic = Magic::from_str(magic_str.trim_start_matches("0x")).unwrap();
```

Comment:
Unrelated to this PR but its pretty surprising that `from_hex` cannot handle the leading `0x`.

Reply:
Not really, `from_hex` is for hashes and such. Those don't use the prefix at all.